[CI] SharedBlobCacheWarmingServiceIT testCacheIsWarmedOnUnhollowing failing

**Build Scans:**
- [elasticsearch-intake #41656 / part2](https://gradle-enterprise.elastic.co/s/vdfvlzee3ouww)
- [elasticsearch-pull-request #153589 / part-2](https://gradle-enterprise.elastic.co/s/qre6bkt5h66yo)
- [elasticsearch-pull-request #153589 / part-2](https://gradle-enterprise.elastic.co/s/nvgkvkxxqxz5m)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:stateless:internalClusterTest" --tests "org.elasticsearch.xpack.stateless.cache.SharedBlobCacheWarmingServiceIT.testCacheIsWarmedOnUnhollowing" -Dtests.seed=F4B3D86CECB83F7F -Dtests.locale=ks-IN -Dtests.timezone=America/Regina -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.stateless.cache.SharedBlobCacheWarmingServiceIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testCacheIsWarmedOnUnhollowing'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: a value greater than <0L>
     but: <0L> was equal to <0L>
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testCacheIsWarmedOnUnhollowing
- [main] 1 consecutive failures in step part2
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 3 failures in test testCacheIsWarmedOnUnhollowing (0.3% fail rate in 882 executions)
- [main] 2 failures in step part-2 (0.3% fail rate in 603 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -593,9 +593,6 @@ tests:
 - class: org.elasticsearch.index.store.BytesReadHeaderIT
   method: testMultiSearchSetsBytesReadHeader
   issue: https://github.com/elastic/elasticsearch/issues/151595
-- class: org.elasticsearch.xpack.stateless.cache.SharedBlobCacheWarmingServiceIT
-  method: testCacheIsWarmedOnUnhollowing
-  issue: https://github.com/elastic/elasticsearch/issues/151600
 - class: org.elasticsearch.datastreams.lifecycle.ExplainDataStreamLifecycleIT
   method: testExplainFailuresLifecycle
   issue: https://github.com/elastic/elasticsearch/issues/151602

--- a/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/cache/SharedBlobCacheWarmingServiceIT.java
+++ b/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/cache/SharedBlobCacheWarmingServiceIT.java
@@ -656,6 +656,10 @@ public class SharedBlobCacheWarmingServiceIT extends AbstractStatelessPluginInte
             metadataContainsReplicatedRanges.set(blobStoreCacheDirectory.metadataContainsReplicatedRanges());
         });
 
+        // Block until warming completes so that the recovery thread's engine.refresh("unhollowing") cannot race
+        // a concurrent cache fill against the warming tasks, which would leave the prewarmed bytes metric at zero.
+        getSharedBlobCacheWarmingService(indexNodeB).setAwaitWarmingForIndexingRecovery(true);
+
         // Trigger unhollowing
         final int moreDocs = randomIntBetween(6, 10);
         logger.info("--> ingesting {} more docs to unhollow the shard", moreDocs);

